app/app.model.js

```diff
@@ -10,17 +11,45 @@ const selectTopics = () => {
     })
 }
 
-const selectArticles = () => {
-    return db.query(`
-        SELECT articles.*,
-        COUNT(comments.comment_id) AS comment_count
-        FROM articles
-        LEFT JOIN
-        comments
-        ON articles.article_id = comments.article_id
-        GROUP BY articles.article_id
-        ORDER BY articles.created_at DESC
-        `)
+const selectArticles = (query) => {
+    const topic = query.topic
+    const sort_by = query.sort_by ?? "created_at"
+    const order = query.order ?? "DESC"
+
+    const sortByGreenList = ["title", "topic", "author", "body", "created_at", "article_img_url"]
+    const orderGreenList = ["ASC", "DESC"]
+
+    const greenLight = sortByGreenList.includes(sort_by) && orderGreenList.includes(order)
+
+    let queryString = format(`
+    SELECT articles.*,
+    COUNT(comments.comment_id) AS comment_count
+    FROM articles
+    LEFT JOIN
+    comments
+    ON articles.article_id = comments.article_id
+    `)
+    
+    if (topic) {
```

Comment:
Try to think of a way to check whether a topic already exists in our database first.